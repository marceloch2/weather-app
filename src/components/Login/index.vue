<template>
  <form class="login" ref="loginForm">
    <fieldset>
      <legend>[ {{title.toUpperCase()}} ]</legend>
      <input
        type="text"
        autofocus
        class="login__username"
        placeholder="username"
        v-model.trim="username"
      >
      <input type="password" class="login__password" placeholder="password" v-model="password">
      <button name="login" type="submit">Login</button>
    </fieldset>
  </form>
</template>

<script>
import Login from "./login";
export default Login;
</script>

<style lang="sass" scoped>
	@import 'login.scss'
</style>
