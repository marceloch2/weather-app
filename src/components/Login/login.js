import { logins } from "../../Utils/constants";
export default {
  name: "Login",
  data() {
    return {
      title: "login",
      username: "marceloch2@gmail.com",
      password: "Demo12345"
    };
  },

  mounted() {
    this.$refs.loginForm.addEventListener("submit", event => {
      event.preventDefault();
      this.login();
    });
  },

  methods: {
    getLoginData() {
      return logins.filter(
        login =>
          login.username === this.username && login.password === this.password
      );
    },
    login() {
      if (this.getLoginData().length) {
        this.$session.start();
        this.$session.set("jwt", this.getLoginData()[0].hash);
        this.$router.push("/home");

        this.$store.commit("setUser", this.getLoginData()[0]);
      }
    }
  }
};
