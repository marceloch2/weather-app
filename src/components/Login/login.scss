.login {
  display: flex;
  justify-content: center;
  align-items: center;

  fieldset {
    border: solid 1px darkslategray;
    padding: 30px;
    width: 400px;
    background-color: rgba(255, 255, 255, 0.8);
  }

  input {
    width: 80%;
    padding: 10px;
    margin-bottom: 15px;
    border: solid 1px darkslategray;
  }

  button {
    padding: 10px;
    width: 50%;
    border: none;
    background-color: darkslategray;
    color: #fff;
  }
}
