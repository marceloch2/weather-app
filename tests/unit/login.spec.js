import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils'
import Login from '@/components/Login'

import createStateConfig from './createStateConfig'

const localVue = createLocalVue()
localVue.use(Vuex)

const stateConfig = createStateConfig()

const store = new Vuex.Store({
  state: stateConfig,
  mutations: {
    setUser: jest.fn()
  }
})

const $router = {
  push: jest.fn()
}

const $session = {
  destroy: () => {
    window.localStorage.removeItem('vue-session-key')
  },
  start: () => jest.fn(),
  set: () => jest.fn()
}

const wrapper = shallowMount(Login, {
  store,
  localVue,
  mocks: {
    $router,
    $session
  }
})

describe('Login.vue', () => {
  beforeEach(() => {})

  it('Snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Can retrieve login information', () => {
    wrapper.vm.username = 'a'
    wrapper.vm.password = 'b'

    expect(wrapper.vm.getLoginData()).toEqual([])

    wrapper.vm.username = 'marceloch2@gmail.com'
    wrapper.vm.password = 'Demo12345'

    expect(wrapper.vm.getLoginData()[0]).toMatchObject({
      username: 'marceloch2@gmail.com',
      password: 'Demo12345'
    })
  })
})
