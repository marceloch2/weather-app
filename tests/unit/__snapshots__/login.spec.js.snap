// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login.vue Snapshot 1`] = `
<form class="login">
  <fieldset>
    <legend>[ LOGIN ]</legend> <input type="text" autofocus="autofocus" placeholder="username" class="login__username"> <input type="password" placeholder="password" class="login__password"> <button name="login" type="submit">Login</button>
  </fieldset>
</form>
`;
